src/renderer/pages/Home.tsx

```diff
@@ -1,11 +1,17 @@
-import NewProjectBlock from 'renderer/components/NewProjectBlock';
-import RecentProjectsBlock from 'renderer/components/RecentProjectsBlock';
+import { useSelector } from 'react-redux';
+import NewProjectBlock from '../components/NewProjectBlock';
+import RecentProjectsBlock from '../components/RecentProjectsBlock';
+import { ApplicationStore } from '../store/helpers';
 
 const HomePage = () => {
+  const hasRecentProjects = useSelector(
+    (store: ApplicationStore) => store.recentProjects.length > 0
+  );
+
   return (
     <>
-      <NewProjectBlock />
-      <RecentProjectsBlock />
+      <NewProjectBlock isFullSize={!hasRecentProjects} />
```

Comment:
Could the home page just get the entire store of recentProjects from redux and then pass it into the <RecentProjectsBlock/> child component? (rather than accessing the store in both the parent and child component)?

Are you looking to avoid re-renders of the entire RecentProjectsBlock component?

Reply:
Yeah the home page only uses it to check if there ARE recent projects, so it only reloads when that changes. If we make the selector load the whole recent projects list, it will re render both. No practically important but slightly nicer implementation this way

Also kind of defeats the point of redux reducing pass-down props for stuff that is in the store